test/io-tests/test_io.py

```diff
@@ -212,13 +218,18 @@ def test_expected_config_from_environment():
 def test_expected_config_from_db_settings(defaultenv):
     "Config should be overriden from database settings"
 
-    env = defaultenv
-    env["PGRST_DB_SETTINGS_PREFIX"] = "instance"
-    expected = (CONFIGSDIR / "db-settings-prefix.config").read_text()
+    expected = (CONFIGSDIR / "expected" / "db-settings-prefix.config").read_text()
+    config = CONFIGSDIR / "db-settings-prefix.config"
 
     # remove db-uri from the config file since it varies
```

Comment:
WDYT about adding the actual db-uri value to the expected output instead? Maybe you can just add a placeholder in the expected file like `<REPLACED_WITH_DB_URI>` and then use a simple replace?

That would make the test code simpler - and would test the "db-uri can't be overwritten from the database", too.

Reply:
> would test the "db-uri can't be overwritten from the database", too.

Hmm.. I don't understand how can this be really proven if I have to replace the expected value.

The `db-uri` I get varies like:

```
db-uri = "postgresql:///postgres?host=/run/user/1000/tmp.MarQgcKD7r/socket&user=postgrest_test_authenticator"
db-uri = "postgresql:///postgres?host=/run/user/1000/tmp.hyIrX2yB1H/socket&user=postgrest_test_authenticator"
db-uri = "postgresql:///postgres?host=/run/user/1000/tmp.YSNO7AmIjZ/socket&user=postgrest_test_authenticator"
```

What I've done instead is to add an additional assert for failing if `db-uri = "postgresql://ignored"`(value from the guc). Which does fail when removing `db-uri` from:

```haskell
          if key `notElem` [
            "server-host", "server-port", "server-unix-socket", "server-unix-socket-mode", "log-level",
            "db-anon-role", "db-uri", "db-channel-enabled", "db-channel", "db-pool", "db-pool-timeout", "db-load-guc-config"]
          then lookup key dbSettings
          else Nothing
```

Additiona

Reply:
Values are first read from the config file (1), then from the environment (2) and then from the database (3), right?

In the IO tests, `db-uri` is read from (2).

Now, we want to test that there is **no** (3) for `db-uri`, because it should not be possible to do that.

We do the following to test that:
- We set `db-uri` to `ignored` in the database
- We assert that the dumped value is still the same value that was provided by the environment variable (2)

This assert is done by comparing the dump vs the expected output. Because `db-uri` is not a static value, but dynamically created, I suggested we just adjust the expected output dynamically by loading the file and then replacing the actual value for `db-uri`.

This should show the same behaviour that you are describing in terms of removing `db-uri` from that list.

Reply:
Or to give a concrete example of what I meant:

```python
expected = expected.replace('<REPLACED_WITH_DB_URI>', defaultenv['PGRST_DB_URI'])
```

If you do that, you only need to assert `dumped == expected` - nothing else.

Reply:
Ahh.. makes sense now!

Thanks for elaborating. I've corrected the test.